### PR TITLE
[SOAR-771] Check Host Block Status with Fortinet Firewall from Microsoft Teams

### PR DESCRIPTION
**Pull request by mhofert-r7**
## Requirements

Workflow contributors, verify you have completed the following items by checking them off:

### Checklist
- [ ] Workflow verified as working in a test
- [ ] Workflow bundle includes `Name_of_Workflow.icon` file, `help.md` file, `workflow.spec.yaml` file, and `screenshots` directory
- [ ] Workflow title should follow general [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Workflow includes a detailed Description
- [ ] Workflow step names follow [Title of Articles](http://grammar.yourdictionary.com/capitalization/rules-for-capitalization-in-titles.html) format
- [ ] Triggers, Decisions, and other workflow steps include descriptions where needed
- [ ] Workflow includes at least one `Final Report` artifact that summarizes workflow execution details
- [ ] Workflow bundle (directory) is named after the title of the workflow in snake case with first letters capitalized, e.g. `workflows/Name_of_Workflow/`
- [ ] `help.md` file includes Description, Key Features, Requirements, Setup, Technical Details, Version History, and References
- [ ] `workflow.spec.yaml` file includes `products`, `name`, `title`, `description`, `use_cases` tags, `keywords` tags, `source_url`, and the `name` for each screenshot matches a PNG image in the `screenshots` directory

### Assessment

Post your results with screenshots in this PR using the following style:

[*] Validating workflow at .

[*] Running Integration Validators...
[*] Executing validator WorkflowDirectoryNameMatchValidator
[*] Executing validator WorkflowFilesValidator
[*] Executing validator WorkflowHelpValidator
[*] Executing validator WorkflowChangelogValidator
[*] Executing validator WorkflowVendorValidator
[*] Executing validator WorkflowVersionValidator
[*] Executing validator WorkflowExtensionValidator
[*] Executing validator WorkflowSupportValidator
[*] Executing validator WorkflowPNGHashValidator
[*] Executing validator WorkflowICONFileNameValidator
[*] Executing validator WorkflowScreenshotValidator
[*] Executing validator WorkflowTitleValidator
[*] Executing validator WorkflowDescriptionValidator
[*] Executing validator WorkflowNameValidator
[*] Executing validator WorkflowProfanityValidator
[*] Executing validator WorkflowHelpPluginUtilizationValidator
[*] Executing validator WorkflowICONFileValidator
[*] Workflow successfully validated!
### Screenshots

Screenshots help your reviewers provide valuable feedback on your contribution. To add a screenshot here, simply drag and drop the PNG files from a file explorer.

#### Workflow Builder

workflow1

#### teams

teams_message1
teams_message2

#### Artifact

artifact1
artifact2

### CODE REVIEW

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams.icon [hunk 1]
```diff
@@ -0,0 +1,1041 @@
...
+  "kom": {
+    "komandVersion": "1.49.5-0-g5c0c6",
+    "komFileVersion": "2.0.0",
+    "exportedAt": "2020-05-07T17:56:39.68091076Z",
+    "workflowVersions": [
+      {
+        "name": "Check Host Block Status with Fortinet Firewall from Microsoft Teams",
+        "type": "runnable",
+        "version": "",
+        "description": "This workflow accepts a Microsoft Teams command, check if an IP address is blocked, and posts if the IP address is blocked.",
```
Comment (jmcadams-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams.icon:11]:
```suggestion
        "description": "This workflow accepts a Microsoft Teams command, checks if an IP address is blocked, and posts the results.",
```

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Description
+
+This workflow accepts a Microsoft Teams command containing an IP address. This IP address will be checked against a pre-defined address group in Fortinet to determine if the IP address is present or not. It is assumed that this address group will be used to store IP addresses for a block policy. A message with the results will be returned.
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md:3]:
```suggestion
This workflow accepts a Microsoft Teams command containing an IP address. This IP address will be checked against a pre-defined address group in Fortinet to determine whether or not the IP address is present. It is assumed that this address group will be used to store IP addresses for a block policy. A message with the results will be returned.
```

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+Sample Microsoft Teams Trigger Commands:
+
+`!block-status 198.51.100.100`
+
+# Key Features
+
+* The ability to see if an IP is currently blocked by the firewall
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md:11]:
```suggestion
* The ability to see if an IP address is currently blocked by the firewall
```

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+# Requirements
+
+* Microsoft Teams connection
+* An admin API key to a Fortigate firewall
```
Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md:16]:
```suggestion
* Admin API key to a Fortinet FortiGate firewall
```

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+# Documentation
+
+## Setup
+
+Import the workflow from the Rapid7 Extension Library and proceed through the Import Workflow wizard in InsightConnect. Import plugins, create or select connections, and rename the workflow as a part of the Import Workflow wizard as necessary.
+
+In the _Check for IP Message_, _IP Blocked Message_, and _IP Not Blocked Message_ steps edit the Team Name and Channel Name input from `change_me` to an appropriate team and channel.
```
Comment (jmcadams-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/help.md:24]:
```suggestion
In the _Check for IP Message_, _IP Blocked Message_, and _IP Not Blocked Message_ steps change the Team Name and Channel Name input from `change_me` to an appropriate team and channel.
```

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml [hunk 6]
```diff
@@ -0,0 +1,27 @@
+extension: workflow
+products: ["insightconnect"]
+name: Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams
+title: "Check Host Block Status with Fortinet Firewall from Microsoft Teams"
+description: This workflow accepts a Microsoft Teams command, check if an IP address is blocked, and posts if the IP address is blocked.
```
Comment (jmcadams-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml:5]: Change to the description from the icon file above.

Comment (mrinehart-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml:5]:
```suggestion
description: Check if an IP address is blocked in Fortinet FortiGate from Microsoft Teams.
```

Comment (mrinehart-r7) [reply workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml:5]: This is a bit more succinct and will fit better in the Hub "tile"

File: workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+name: Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams
+title: "Check Host Block Status with Fortinet Firewall from Microsoft Teams"
+description: Check if an IP address is blocked in Fortinet FortiGate from Microsoft Teams.
+version: 1.0.0
+vendor: rapid7
+support: rapid7
+status: []
+hub_tags:
+  use_cases: [remediation_management,threat_detection_and_response]
+  keywords: [fortinet ,firewall, microsoft_teams, chatops, work_from_home]
```
Comment (jschipp-r7) [inline workflows/Check_Host_Block_Status_with_Fortinet_Firewall_from_Microsoft_Teams/workflow.spec.yaml:12]:
```suggestion
  keywords: [fortinet, firewall, microsoft_teams, chatops, work_from_home]
```

**Pull request merged**